coding/mmap_reader.cpp

```diff
@@ -21,8 +24,35 @@ class MmapReader::MmapData
 public:
   explicit MmapData(std::string const & fileName, Advice advice)
   {
-    // @TODO add windows support
-#ifndef OMIM_OS_WINDOWS
+#ifdef OMIM_OS_WINDOWS
+    m_hFile = CreateFileA(fileName.c_str(), GENERIC_READ, 0, nullptr, OPEN_EXISTING, FILE_ATTRIBUTE_NORMAL, nullptr);
+    if (m_hFile == INVALID_HANDLE_VALUE)
+      MYTHROW(Reader::OpenException, ("Can't open file:", fileName, "win last error:", GetLastError()));
+
+    m_hMapping = CreateFileMappingA(m_hFile, nullptr, PAGE_READONLY, 0, 0, nullptr);
+    if (m_hMapping == nullptr)
+    {
+      CloseHandle(m_hFile);
```

Comment:
Please use `base/scope_guard.h` to avoid copy-paste, and `.release()` it in the success case.